package_nso_to_oc/xe/xe_routing_policy.py

```diff
@@ -0,0 +1,521 @@
+            if not "expr" in entry:
+                continue
+            if entry["expr"].startswith("deny"):
+                all_processed = False
+                routing_policy_notes.append(
+f"""
+Community Name: {community.get("name")}
+Community Type: {type}
+Entry: {entry["expr"]}
+This entry contains a deny operation, which is not supported in OpenConfig. Translation to OC was skipped.
+""")
+                continue
+
+            member = entry["expr"][entry["expr"].find(' ') + 1:]
+            members.append(well_known_members.get(member, member))
+
+            # Ensure the value we're nullifying does exist
+            if common.get_index_or_default(entry_after, entry_index, None):
+                entry_after[entry_index] = None
+
+        community_sets["openconfig-bgp-policy:community-sets"]["openconfig-bgp-policy:community-set"].append(new_community_set)
+
+        if all_processed:
+            common.get_index_or_default(community_list_after.get(type, []), community_index, {})["name"] = None
+
+def process_ext_community_sets(config_before, config_after):
+    ext_community_sets = {"openconfig-bgp-policy:ext-community-sets": {"openconfig-bgp-policy:ext-community-set": []}}
+    ext_community_list = config_before.get("tailf-ned-cisco-ios:ip", {}).get("extcommunity-list", {})
+    ext_community_list_after = config_after.get("tailf-ned-cisco-ios:ip", {}).get("extcommunity-list", {})
+    process_ext_community_members(ext_community_sets, "standard", ext_community_list, ext_community_list_after)
+    process_ext_community_members(ext_community_sets, "expanded", ext_community_list, ext_community_list_after)
+    openconfig_routing_policies["openconfig-routing-policy:routing-policy"]["openconfig-routing-policy:defined-sets"]["openconfig-bgp-policy:bgp-defined-sets"].update(ext_community_sets)
+
+def process_ext_community_members(ext_community_sets, type, ext_community_list, ext_community_list_after):
+    all_processed = True
+
+    for ext_community_index, ext_community in enumerate(ext_community_list.get(type, {"no-mode-list": []}).get("no-mode-list", [])):
+        ext_new_community_set = {
+            "openconfig-bgp-policy:ext-community-set-name": ext_community.get("name"),
+            "openconfig-bgp-policy:config": {
+                "openconfig-bgp-policy:ext-community-set-name": ext_community.get("name"),
+                "openconfig-bgp-policy:ext-community-member": []
+            }
+        }
+        ext_members = ext_new_community_set["openconfig-bgp-policy:config"]["openconfig-bgp-policy:ext-community-member"]
+        entry_after = common.get_index_or_default(ext_community_list_after.get(type, {"no-mode-list": []}).get("no-mode-list", []), ext_community_index, {}).get("entry", [])
+
+        for entry_index, entry in enumerate(ext_community.get("entry", [])):
+            if not "expr" in entry:
+                continue
+            if entry["expr"].startswith("deny"):
```

Comment:
If deny, we must note, keep processing for other notes, but can't appended to OC.